src/lib/state/actions/deposit.js

```diff
@@ -74,3 +75,56 @@ export const discard = (event, formik) => {
     });
   };
 };
+
+/**
+ * Reserve the PID after having saved the current draft
+ * @param {string} pidType - the PID type to reserve the PID for
+ * @param {object} formik- formik object
+ */
+export const reservePID = (pidType, formik) => {
+  return async (dispatch, getState, config) => {
+    const controller = config.controller;
+
+    dispatch({
+      type: RESERVE_PID_STARTED,
+    });
+
+    // PIDS-FIXME: this is not the latest version of the form values,
+    // it will replace the form values with what was in the record
+    const draft = getState().deposit.record;
```

Comment:
@zzacharo I have no idea how to solve this. Using `submitFormData` instead will not wait for the state to be updated. We can chat about this is you have ideas.